assets/js/editor/Editor.tsx

```diff
@@ -99,23 +100,39 @@ export default function Editor({ source, adaptor, onChange }: EditorProps) {
       editor.executeEdits("snippets", [op]);
 
       // Ensure the snippet is fully visible
-      const newLastLine = editor.model.getLineCount();
+      const newLastLine = model.getLineCount();
       editor.revealLines(lastLine + 1, newLastLine, 0) // 0 = smooth scroll
 
       // Set the selection to the start of the snippet
       editor.setSelection(new monaco.Range(lastLine+1, 0, lastLine+1, 0));
       
       // ensure the editor has focus
       editor.focus();
-    }
+    };
+
+    document.addEventListener('insert-snippet', listeners.current.insertSnippet);
+  }, []);
+
+  useEffect(() => {
+    // Create a node to hold overflow widgets
+    // This needs to be at the top level so that tooltips clip over Lightning UIs
+    const overflowNode = document.createElement('div');
+    overflowNode.className = "monaco-editor widgets-overflow-container";
+    document.body.appendChild(overflowNode);
+
+    setOptions({
+      ...defaultOptions,
+      overflowWidgetsDomNode: overflowNode,
+      fixedOverflowWidgets: true
+    })
 
-    document.addEventListener('insert-snippet', handleInsertSnippet)
-    
     return () => {
-      document.removeEventListener(handleInsertSnippet);
+      overflowNode.parentNode?.removeChild(overflowNode);
+      if (listeners.current?.insertSnippet) {
+        document.removeEventListener('insert-snippet', listeners.current.insertSnippet);
```

Comment:
The code to remove the insert-snippet was in completely the wrong place. Strange this, it should have never worked. I have a feeling there's been a funny merge conflit.

Anyway, if you wanted to test this, you can add some logging to the listener and mount/unmount the editor a few times (by showing/dismissing the panel) and you'll see that it only ever triggers one insert-snippet event.